File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
    string normalize(const string& input){
        string result;
        for(char ch:input){
            if(std::isalnum(static_cast<unsigned char>(ch))){
                result+=tolower(ch);
            }
        }
        return result;
    }
public:
    bool isPalindrome(string s) {
        string ss = normalize(s);
        int l=0, r=ss.length()-1;
        while(l<r){
            if(ss[l]!=ss[r]){
                return false;
            }
            l++;r--;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.go
func isPalindrome(s string) bool {
    l, r := 0, len(s)-1

    for l < r {
        if !unicode.IsDigit(rune(s[l])) &&
            !unicode.IsLetter(rune(s[l])) {
            l++
            continue
        }

        if !unicode.IsDigit(rune(s[r])) &&
            !unicode.IsLetter(rune(s[r])) {
            r-- 
            continue
        }

        if unicode.ToLower(rune(s[l])) != unicode.ToLower(rune(s[r])) {
            return false
        }

        l++
        r--
    }

    return true
}